server/src/test/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryTests.java

```diff
@@ -454,7 +416,7 @@ protected void snapshotFile(SnapshotShardContext context, BlobStoreIndexShardSna
             files.add(ShardSnapshotTaskRunnerTests.dummyFileInfo());
         }
         repository.snapshotFiles(context, files, allFilesUploadListener);
-        listenerCalled.get();
+        listenerCalled.get(10, TimeUnit.SECONDS);
```

Comment:
It seems we could use a helper method for this similar to `safeAwait` etc.

Reply:
Yeah or we could just make this a `CountDownLatch`. Didn't really want to make too many changes here.